### ISSUE TITLE
rf_event_service.py : -et/--eventtypes is not working eventhough its provided

### ISSUE THREAD

**Issue by rajeshkumarmhpe** [DMTF/Redfish-Tacklebox#89]
Our Server machine supports EventTypes which is working fine using curl. But its not working from the tacklebox tool script. 

# rf_event_service.py --user administrator --password HPE1nnov --rhost https://x.x.x.x  subscribe -dest https://y.y.y.y -et [Alert ResourceAdded ResourceRemoved ResourceUpdated StatusChange] -c 1 
Operation failed: HTTP 400
A general error has occurred.  See Resolution for information on how to resolve the error, or @Message.ExtendedInfo if Resolution is not provided.
The property EventTypes is a required property and must be included in the request.

**Tried all these options but none of them working:**
-et [Alert ResourceAdded ResourceRemoved ResourceUpdated StatusChange]
-et [Alert,ResourceAdded,ResourceRemoved,ResourceUpdated,StatusChange]
-et ["Alert" "ResourceAdded" "ResourceRemoved" "ResourceUpdated" "StatusChange"]
-et ["Alert","ResourceAdded","ResourceRemoved","ResourceUpdated","StatusChange"]
-et ["Alert"]
-et [Alert]
-et Alert
-et "Alert"

Let us know whats the problem around here.

**Comment by mraineri** [comment]
Can you please confirm the version of Tacklebox installed? Prior to version 1.2.0, "EventTypes" was not being supplied to the service properly.

You can check with `pip show redfish_utilities`. If you have an older version of Tacklebox, you can upgrade with `pip install redfish_utilities -U`

**Comment by rajeshkumarmhpe** [comment]
# pip3 show redfish_utilities
Name: redfish-utilities
Version: 1.2.8
Summary: Redfish Utilities
Home-page: https://github.com/DMTF/Redfish-Tacklebox
Author: DMTF, https://www.dmtf.org/standards/feedback
Author-email: None
License: BSD 3-clause "New" or "Revised License"
Location: /usr/local/lib/python3.6/site-packages
Requires: redfish, XlsxWriter

**Comment by mraineri** [comment]
Thanks, debugging now and I'm finding another issue that;s in the script portion.

**Comment by mraineri** [comment]
I have a fix in a branch; it should be reviewed by others and approved later today

**Comment by rajeshkumarmhpe** [comment]
I changed my side script based on your fix in #90 .But still fails.

**Comment by rajeshkumarmhpe** [comment]
Need to know what is the exact syntax which needs to be parsed for -et/--eventtypes 
whether this is correct :
-et [Alert, ResourceChanged]

**Comment by rajeshkumarmhpe** [comment]
> If you do have 1.2.0 or higher, can you also try this curl command? Tacklebox should be performing the same operation as this command.
> 
> `curl -k -u administrator:HPE1nnov -X POST "https://x.x.x.x/redfish/v1/EventService/Subscriptions" -d '{"Destination": "https://y.y.y.y", "Context": "1", "Protocol": "Redfish", "EventTypes": [ "Alert" ] }' -H "Content-Type: application/json"`

[Rajesh]This is working fine

**Comment by mraineri** [comment]
You don't specify an array in the argument; it's just a list of event types. For example:

`-et Alert StatusChanged ResourceAdded`

**Comment by rajeshkumarmhpe** [comment]
Still the same 

# rf_event_service.py --user administrator --password HPE1nnov --rhost https://x.x.x.x subscribe -dest https://y.y.y.y/redfish_event_logger -et Alert ResourceAdded ResourceChanged  -c 1
Operation failed: HTTP 400
A general error has occurred.  See Resolution for information on how to resolve the error, or @Message.ExtendedInfo if Resolution is not provided.
The property EventTypes is a required property and must be included in the request.

**Comment by rajeshkumarmhpe** [comment]
I too changed your fix in the script locally:

try:
    if args.command == "subscribe":
        response = redfish_utilities.create_event_subscription( redfish_obj, args.destination, format = args.format, client_context = args.context,
            expand = args.expand, resource_types = args.resourcetypes, registries = args.registries, event_types = args.eventtypes )
        print( "Created subscription '{}'".format( response.getheader( "Location" ) ) )

**Comment by rajeshkumarmhpe** [comment]
I ran from where i cloned the redfish-tacklebox
/root/Redfish-Tacklebox/scripts/rf_event_service.py

**Comment by rajeshkumarmhpe** [comment]
I got your point.
This is working fine.
once your code is merged i will use it.Thanks
